packages/x-tree-view/src/internals/models/plugin.ts

```diff
@@ -154,14 +199,42 @@ export type TreeItemWrapper<TSignatures extends readonly TreeViewAnyPluginSignat
 export type TreeRootWrapper = (params: { children: React.ReactNode }) => React.ReactNode;
 
 export type TreeViewPlugin<TSignature extends TreeViewAnyPluginSignature> = {
+  /**
+   * The main function of the plugin that will be executed by the tree-view.
```

Comment:
I'm always getting this naming thing wrong, but I think it's
```suggestion
   * The main function of the plugin that will be executed by the Tree View.
```